### PR TITLE
Store deleted documents in the kdb index.

### PR DESCRIPTION
**Pull request by noffle**
This is a prerequisite for the larger goal of exposing deletions through the osm-p2p-db API.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -43,7 +43,8 @@ function DB (opts) {
       if (v && v.lat !== undefined && v.lon !== undefined) {
         next(null, { type: 'put', point: ptf(v) })
       } else if (d && Array.isArray(row.value.points)) {
-        next(null, { type: 'del', points: row.value.points.map(ptf) })
+        var pt = row.value.points.map(ptf)[0]
```
Comment (gmaclennan) [inline index.js]: We need an index record for each point in `row.value.points`. This property is set in the delete code and is an array of the lat/lon of each parent. Each of those records should be updated to point to the deleted record, i.e. we need two `put`s

File: index.js [hunk 2]
```diff
@@ -308,6 +309,10 @@ DB.prototype._onpt = function (pt, seen, cb) {
   self.log.get(link, function (err, doc) {
     if (doc && doc.value && doc.value.k && doc.value.v) {
       addDoc(doc.value.k, link, doc.value.v)
+    } else if (doc && doc.value && doc.value.d && doc.value.points) {
+      var pt = doc.value.points[0]
+      pt.deleted = true
+      addDoc(doc.value.d, link, pt)
```
Comment (gmaclennan) [inline index.js]: I don't quite understand what this block is doing, can you explain?

Comment (noffle) [inline index.js]: I'm marking deleted documents as such, but (below) filtering them out. This means no API changes or change in behaviour, but the new logic is in place. This is to make the breaking change PR smaller and a bit easier to reason about.

File: index.js (hunk 1 above)
Comment (noffle) [inline index.js]: Yes, you are 100% right; good catch.

**Pull request closed without merging**